test/filter-test.js

```diff
@@ -0,0 +1,263 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const FullNode = require('../lib/node/fullnode');
+const {filters} = require('../lib/blockstore/common');
+const {BufferSet} = require('buffer-map');
+const {opcodes} = require('../lib/script/common');
+const FilterIndexer = require('../lib/indexer/filterindexer');
+const Block = require('../lib/primitives/block');
+const Golomb = require('../lib/golomb/golomb');
+const {U64} = require('n64');
+const BasicFilter = require('../lib/golomb/basicFilter');
+
+class TestFilter extends Golomb {
+  constructor() {
+    super(21, new U64(744931));
+  }
+}
+
+class TestBlock extends Block {
+  constructor() {
+    super();
+  }
+
+  toTestFilter(view) {
+    const hash = this.hash();
+    const key = hash.slice(0, 16);
+    const items = new BufferSet();
+
+    for (let i = 0; i < this.txs.length; i++) {
+    const tx = this.txs[i];
+
+    for (const output of tx.outputs) {
+        if (output.script.length === 0)
+        continue;
+
+        // In order to allow the filters to later be committed
+        // to within an OP_RETURN output, we ignore all
+        // OP_RETURNs to avoid a circular dependency.
+        if (output.script.raw[0] === opcodes.OP_RETURN)
+        continue;
+
+        items.add(output.script.raw);
+    }
```

Comment:
nit: the `for...` line and closing `}` should be indented once more each. Same with the for loop on lines 51-56